From 02437431e5c3cc69bdea6e9bc7c6a78ab96a487b Mon Sep 17 00:00:00 2001
From: Damian Terentiev <infusiastic@gmail.com>
Date: Wed, 11 Feb 2009 23:09:45 +0800
Subject: [PATCH] Fixed the template runner gem method to output :lib => false correctly

Fixed the template runner so that gem method with :lib => false option
now outputs :lib => false instead of :lib => 'false'
---
 .../generators/applications/app/template_runner.rb |    2 +-
 .../test/generators/rails_template_runner_test.rb  |    5 +++++
 2 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails_generator/generators/applications/app/template_runner.rb b/railties/lib/rails_generator/generators/applications/app/template_runner.rb
index 84e36ec..c7074d6 100644
--- a/railties/lib/rails_generator/generators/applications/app/template_runner.rb
+++ b/railties/lib/rails_generator/generators/applications/app/template_runner.rb
@@ -89,7 +89,7 @@ module Rails
       gems_code = "config.gem '#{name}'"
 
       if options.any?
-        opts = options.inject([]) {|result, h| result << [":#{h[0]} => '#{h[1]}'"] }.sort.join(", ")
+        opts = options.inject([]) {|result, h| result << [":#{h[0]} => #{h[1].inspect.gsub('"',"'")}"] }.sort.join(", ")
         gems_code << ", #{opts}"
       end
 
diff --git a/railties/test/generators/rails_template_runner_test.rb b/railties/test/generators/rails_template_runner_test.rb
index fcc0206..bed7563 100644
--- a/railties/test/generators/rails_template_runner_test.rb
+++ b/railties/test/generators/rails_template_runner_test.rb
@@ -82,6 +82,11 @@ class RailsTemplateRunnerTest < GeneratorTestCase
     assert_rails_initializer_includes("config.gem 'mislav-will-paginate', :lib => 'will-paginate', :source => 'http://gems.github.com'")
   end
 
+  def test_gem_with_lib_option_set_to_false_should_put_gem_dependency_in_enviroment_correctly
+    run_template_method(:gem, 'mislav-will-paginate', :lib => false, :source => 'http://gems.github.com')
+    assert_rails_initializer_includes("config.gem 'mislav-will-paginate', :lib => false, :source => 'http://gems.github.com'")
+  end
+
   def test_environment_should_include_data_in_environment_initializer_block
     load_paths = 'config.load_paths += %w["#{RAILS_ROOT}/app/extras"]'
     run_template_method(:environment, load_paths)
-- 
1.6.1.3

